cypress/e2e/orders/edit-fund-and-decrease-cost-in-po-line-for-paid-invoice.cy.js

```diff
@@ -0,0 +1,259 @@
+    },
+  };
+  const budgets = {
+    first: {
+      ...Budgets.getDefaultBudget(),
+      allocated: 100,
+      fiscalYearId: fiscalYear.id,
+      fundId: funds.first.id,
+    },
+    second: {
+      ...Budgets.getDefaultBudget(),
+      allocated: 100,
+      fiscalYearId: fiscalYear.id,
+      fundId: funds.second.id,
+    },
+  };
+  const testData = {
+    organization: NewOrganization.getDefaultOrganization(),
+    fiscalYear,
+    user: {},
+  };
+
+  before('Create test data', () => {
+    cy.getAdminToken().then(() => {
+      FiscalYears.createViaApi(fiscalYear);
+      Ledgers.createViaApi(ledger);
+      Object.values(funds).forEach((fund) => {
+        Funds.createViaApi(fund);
+      });
+      Object.values(budgets).forEach((budget) => {
+        Budgets.createViaApi(budget);
+      });
+
+      Organizations.createOrganizationViaApi(testData.organization).then(() => {
+        testData.order = {
+          ...NewOrder.getDefaultOrder({ vendorId: testData.organization.id }),
+          reEncumber: true,
+        };
+        testData.orderLine = BasicOrderLine.getDefaultOrderLine({
+          listUnitPrice: 50,
+          fundDistribution: [{ code: funds.first.code, fundId: funds.first.id, value: 100 }],
+        });
+
+        Orders.createOrderWithOrderLineViaApi(testData.order, testData.orderLine).then((order) => {
+          testData.order = order;
+
+          Orders.updateOrderViaApi({
+            ...testData.order,
+            workflowStatus: ORDER_STATUSES.OPEN,
+          });
+
+          OrderLines.getOrderLineViaApi({ query: `poLineNumber=="*${order.poNumber}*"` }).then(
+            (orderLines) => {
+              testData.orderLine = orderLines[0];
+
+              Invoices.createInvoiceWithInvoiceLineViaApi({
+                vendorId: testData.organization.id,
+                fiscalYearId: testData.fiscalYear.id,
+                poLineId: testData.orderLine.id,
+                fundDistributions: testData.orderLine.fundDistribution,
+                accountingCode: testData.organization.erpCode,
+                subTotal: 50,
+                releaseEncumbrance: false,
+              }).then((invoice) => {
+                testData.invoice = invoice;
+
+                Invoices.changeInvoiceStatusViaApi({
+                  invoice: testData.invoice,
+                  status: INVOICE_STATUSES.PAID,
+                });
+              });
+            },
+          );
+        });
+      });
+    });
+
+    cy.createTempUser([
+      Permissions.uiFinanceViewFundAndBudget.gui,
+      Permissions.uiInvoicesCanViewInvoicesAndInvoiceLines.gui,
+      Permissions.uiOrdersEdit.gui,
+    ]).then((userProperties) => {
+      testData.user = userProperties;
+
+      cy.login(testData.user.username, testData.user.password, {
+        path: TopMenu.ordersPath,
+        waiter: Orders.waitLoading,
+      });
+    });
+  });
+
+  after('Delete test data', () => {
+    cy.getAdminToken().then(() => {
```

Comment:
Nedd to delelte all items from before section

Reply:
When Invoice is moved to Paid status system creates transaction and this disallows to delete all related objects - Invoice, Order, Budget, Fund, Ledger and FY